### PR TITLE
Make new trust / key store properties optional

### PR DESCRIPTION
**Pull request by brudo**
Make new trust / key store properties optional in the inbound transport, and handle at run time if they are not set or are missing altogether, for compatibility with previously configured inputs that did not have these properties.

### CODE REVIEW

**Comment by brudo** [comment]
@hanoch please review this - it provides forward compatibility when updating from a version of the component before PR #9 to any later revision. Without this, errors appear in the log, indicating that a required property was not found. If you recreate the input after updating the transport, it does add the missing properties, but this extra step should not be needed (and the error message in the log seems misleading since these properties are normally not required). Thanks!

File: activemq-transport/src/main/java/com/esri/geoevent/transport/activemq/ActiveMQInboundTransportDefinition.java [hunk 1]
```diff
@@ -54,10 +54,10 @@ public ActiveMQInboundTransportDefinition() {
       propertyDefinitions.put("artemisKSPath", new PropertyDefinition("ARTEMIS_KS_PATH", PropertyType.String, null, "${com.esri.geoevent.transport.activemq-transport.TRANSPORT_IN_ARTEMIS_KS_PATH_LBL}", "${com.esri.geoevent.transport.activemq-transport.TRANSPORT_IN_ARTEMIS_KS_PATH_DESC}", true, false));
       propertyDefinitions.put("artemisTSPasswd", new PropertyDefinition("ARTEMIS_TS_PASSWD", PropertyType.String, null, "${com.esri.geoevent.transport.activemq-transport.TRANSPORT_IN_ARTEMIS_TS_PASSWD_LBL}", "${com.esri.geoevent.transport.activemq-transport.TRANSPORT_IN_ARTEMIS_TS_PASSWD_DESC}", true, false));
       propertyDefinitions.put("artemisKSPasswd", new PropertyDefinition("ARTEMIS_KS_PASSWD", PropertyType.String, null, "${com.esri.geoevent.transport.activemq-transport.TRANSPORT_IN_ARTEMIS_KS_PASSWD_LBL}", "${com.esri.geoevent.transport.activemq-transport.TRANSPORT_IN_ARTEMIS_KS_PASSWD_DESC}", true, false));*/
```
Comment (hanoch) [inline activemq-transport/src/main/java/com/esri/geoevent/transport/activemq/ActiveMQInboundTransportDefinition.java]: please remove this commented-out section.

**Review by hanoch — CHANGES_REQUESTED** [review]
Overall looks okay, following is a cleanup request:

**Comment by brudo** [comment]
@hanoch I have removed the commented-out old definitions. Thanks for spotting that!

File: activemq-transport/src/main/java/com/esri/geoevent/transport/activemq/ActiveMQInboundTransportDefinition.java (hunk 1 above)
Comment (brudo) [reply activemq-transport/src/main/java/com/esri/geoevent/transport/activemq/ActiveMQInboundTransportDefinition.java]: Done

**Review by hanoch — APPROVED** [review]

**Comment by hanoch** [comment]
merged.

**Pull request merged**